src/interop.go.md

```diff
@@ -0,0 +1,52 @@
+# Go interop
+
+`Nim` and `go` are both statically typed, compiled languages capable of interop via a simplifed C ABI, making interop between the two languages simple.
+
+## Threads
+
+`go` includes a native `M:N` scheduler for running `go` tasks - because of this, care must be taken both when calling Nim code from `go`: the thread from which the call will happen is controlled by `go` and  we must initialise the Nim garbage collector in every function exposed to go:
+
+```nim
+# Export our function to go
+proc f() {.exportc.} =
+  when declared(setupForeignThreadGc): setupForeignThreadGc()
```

Comment:
That merits a specific example indeed, though in this case it seems that the technique below would be better suited, ie transfer the content to a single nim thread - running with `GOMAXPROCS=1` sounds like an unrealistic option for most use cases, since go is built from ground up around M:N threading.

Reply:
Well, yes, my comment meant to point out that the advice will lead you to problems. Currently, the text presents it as a solution.

Reply:
f1d5c84be608c8a3289e25cb6b90787b0123f4d9 expands on this section to point out the cross-thread variable problems - adding a simple `MyAPI` pointer solves a few of these problems: forcing the user to call an "init" function early on and giving the code a natural place for _some_ data: the problem of course is that this type must _not_ use any GC types - not sure yet how to best phrase this / enforce it with code structure.